Haskell/syntax_test_haskell.hs

```diff
@@ -3,6 +3,9 @@
 23*36  -- single line comment
 --     ^^ punctuation.definition.comment.haskell
 --     ^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
+23*36  --------------------------------------------------- single line comment
+--     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ punctuation.definition.comment.haskell
+--     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
```

Comment:
The `^` markers should propably span the whole comment (token to test).

Reply:
Could you submit this as a code change suggestion?

Reply:
The followng snippet contains a bit more accurate tests and more than requested, but I'd use ...

```
23*36  --------------------------------------------------- single line comment
--    ^ - comment - punctuation
--     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell punctuation.definition.comment.haskell
--                                                        ^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell - punctuation
```

... to also ensure boundaries are correct.

Reply:
Is this what you mean? If so, happy to apply the change and push it. Tests all passing:

```diff
@@ -4,10 +4,12 @@
 --     ^^ punctuation.definition.comment.haskell
 --     ^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
 23*36  --------------------------------------------------- single line comment
 --     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ punctuation.definition.comment.haskell
 --     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
+--    ^ - comment - punctuation
+--     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell punctuation.definition.comment.haskell
 23*36
 -- <- - comment.line.double-dash.haskell
 
    {- block comment -} 23*36
 -- ^^ punctuation.definition.comment.begin.haskell
```

```
Success: 1270 assertions in 1 files passed
[Finished]
```

Reply:
Or, perhaps this is all we need:

```diff
@@ -2,12 +2,11 @@
 
 23*36  -- single line comment
 --     ^^ punctuation.definition.comment.haskell
 --     ^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
 23*36  --------------------------------------------------- single line comment
---     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ punctuation.definition.comment.haskell
---     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell
+--     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ comment.line.double-dash.haskell punctuation.definition.comment.haskell
 23*36
 -- <- - comment.line.double-dash.haskell
 
    {- block comment -} 23*36
 -- ^^ punctuation.definition.comment.begin.haskell
```

Reply:
The 3 lines of my post replace those red lines in your last one.

Reply:
Done. Thank you.